lib/url.c

```diff
@@ -4028,6 +4034,14 @@ static CURLcode create_conn(struct Curl_easy *data,
    *************************************************************/
   result = resolve_server(data, conn, async);
 
+  /* Strip trailing dots. */
+  strip_trailing_dot(&conn->host);
+  if(conn->bits.httpproxy)
+    strip_trailing_dot(&conn->http_proxy.host);
+  if(conn->bits.socksproxy)
+    strip_trailing_dot(&conn->socks_proxy.host);
+  // not dot-stripping: conn->conn_to_host->name
```

Comment:
checksrc will complain on this use of C++/C99 comment. Go with /* comment */.

Try `make checksrc` and you'll find a few other nits.

Reply:
Done.